include/fi_log.h

```diff
@@ -35,32 +35,107 @@
 #define FI_LOG_H
 
 #if HAVE_CONFIG_H
-#  include <config.h>
+#include <config.h>
 #endif /* HAVE_CONFIG_H */
 
-extern int fi_log_level;
+#include <fi_list.h>
+#include <stdbool.h>
+#include <stdio.h>
+
+/*
+ * Define the basic subsystems that can be used to enable fine grained logging.
+ */
+#define FABRIC (0)
+#define DOMAIN (1)
+#define CM (2)
+#define EP (3)
+#define AV (4)
+#define CQ (5)
+#define EQ (6)
+#define MR (7)
+
+#define LOG_WARN (1)
+#define LOG_TRACE (3)
+#define LOG_INFO (7)
+
+#define FABRIC (0)
+#define PSMX_HANDLE (1)
+#define SOCK_HANDLE (2)
+#define VERBS_HANDLE (3)
+#define USNIC_HANDLE (4)
+#define DYNAMIC_START (5)
```

Comment:
If you're going to have a static set of values, it might be good to give yourself some headroom between the last defined value and the beginning of "dynamic" space for forward compatibility / ABI reasons.